ui/webui/resources/polymer_overriding.js

```diff
@@ -1,33 +1,125 @@
+  const allBraveTemplateModificationsMap = {}
+
+  function addBraveTemplateModifications(component) {
+    if (allBraveTemplateModificationsMap[component.is]) {
+      const moduleName = component.is
+      const domModule = Polymer.DomModule.import(moduleName)
+      if (domModule) {
+        const template = domModule.querySelector('template')
+        if (template) {
+          const templateContent = template.content
+          const t0 = performance.now()
+          allBraveTemplateModificationsMap[component.is](templateContent)
+          const t1 = performance.now()
+          console.debug(`Modifying template '${component.is}' took ${t1 - t0}ms`)
+        }
+      }
+      delete allBraveTemplateModificationsMap[component.is]
+    }
+  }
+
+  function addBraveStyleOverrides(component) {
+    // does have template style element?
+    const moduleName = component.is
+    const domModule = Polymer.DomModule.import(moduleName)
+    if (!domModule) {
+      return
+    }
+    // has override?
+    const overrideModuleName = `brave-override-style-${moduleName}`
+    const hasOverride = !!Polymer.DomModule.import(overrideModuleName)
+    if (!hasOverride) {
+      return
+    }
+    const template = domModule.querySelector('template')
+    if (!template) {
+      return
+    }
+    const styleElement = template.content.querySelector('style')
+    if (!styleElement) {
+      return
+    }
+    styleElement.setAttribute(
+      'include',
+      `${styleElement.getAttribute('include')} ${overrideModuleName}`
+    )
+  }
+
+  function RegisterPolymerComponentBehaviors(behaviorsMap) {
     Object.assign(allBehaviorsMap, behaviorsMap)
   }
 
+  function RegisterPolymerTemplateModifications(modificationsMap) {
+    Object.assign(allBraveTemplateModificationsMap, modificationsMap)
+  }
+
+  function OverrideIronIcons(iconSetName, overridingIconSetName, iconOverrides) {
+    // Here, we change the underlying DOM.
+    // We cannot simply change `IconSet._icons` since
+    // there are occasions where this will get re-parsed from DOM elements.
+    const meta = Polymer.Base.create('iron-meta', {type: 'iconset'})
+    const srcIconSet = meta.byKey(iconSetName)
+    const overrideIconSet = meta.byKey(overridingIconSetName)
+    for (const chromiumIconName in iconOverrides) {
+      const chromiumIcon = srcIconSet.querySelector(`#${chromiumIconName}`)
+      if (!chromiumIcon) {
+        console.error(`[brave overrides] Could not find chromium icon '${chromiumIconName}' in iconset '${iconSetName}' for replacement!`)
+        continue
+      }
+      const braveIconName = iconOverrides[chromiumIconName]
+      const braveIcon = overrideIconSet.querySelector(`#${braveIconName}`)
+      if (!braveIcon) {
+        console.error(`[brave overrides] Could not find brave icon '${braveIconName}' in iconset '${overridingIconSetName}' for replacement!`)
+        continue
+      }
+      // replace
+      chromiumIcon.innerHTML = braveIcon.innerHTML
```

Comment:
same as above, shouldn't this be in an `else` to avoid an error?

Reply:
The `continue` on L107 avoids execution here

Reply:
ah, didn't notice that at first